lib/src/util/logging.cc

```diff
@@ -3,44 +3,134 @@
 #define LEATHERMAN_LOGGING_NAMESPACE "puppetlabs.pcp_client.configuration"
 #include <leatherman/logging/logging.hpp>
 
+#include <boost/log/core.hpp>
+#include <boost/log/sinks/basic_sink_backend.hpp>
+#include <boost/log/sinks/sync_frontend.hpp>
+#include <boost/log/sources/logger.hpp>
+#include <boost/log/sources/severity_logger.hpp>
+#include <boost/log/sources/record_ostream.hpp>
+#include <boost/log/utility/setup/common_attributes.hpp>
+#include <boost/log/attributes/scoped_attribute.hpp>
+
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/date_time/gregorian/gregorian.hpp>
+
+#include <boost/smart_ptr/make_shared_object.hpp>
+
 #include <map>
+#include <fstream>
+#include <cstdint>
 
 namespace PCPClient {
 namespace Util {
 
 namespace lth_log = leatherman::logging;
+namespace sinks = boost::log::sinks;
+namespace expr = boost::log::expressions;
+namespace attrs = boost::log::attributes;
+
+//
+// PCP Access Logging - sink backend
+//
+
+BOOST_LOG_ATTRIBUTE_KEYWORD(access_outcome, "AccessOutcome", std::string)
+
+class access_writer : public sinks::basic_sink_backend<sinks::synchronized_feeding>
+{
+  public:
+    explicit access_writer(std::shared_ptr<std::ostream> sink_stream_ptr);
+    void consume(boost::log::record_view const& rec);
+
+  private:
+    std::shared_ptr<std::ostream> _sink_stream_ptr;
+};
+
+access_writer::access_writer(std::shared_ptr<std::ostream> sink_stream_ptr)
+    : _sink_stream_ptr {std::move(sink_stream_ptr)}
+{
+}
+
+void access_writer::consume(boost::log::record_view const &rec)
+{
+    auto timestamp = boost::log::extract<boost::posix_time::ptime>("TimeStamp", rec);
+    auto message = rec[expr::smessage];
 
-static void setupLoggingImp(std::ostream &stream,
+    *_sink_stream_ptr << '[' << boost::gregorian::to_iso_extended_string(timestamp->date())
+                      << ' ' << boost::posix_time::to_simple_string(timestamp->time_of_day())
+                      << "] " << *message << std::endl;
+}
+
+//
+// PCP Access Logging - logger
+//
+
+void doLogAccess(std::string const& message,
+                 int line_num)
+{
+    // TODO(ale): consider adding an option to prevent access log
```

Comment:
Seems reasonable to have a flag for logging access. Having it interdependent with log level could be confusing (which is a separate issue; maybe we don't want a severity logger).